OrbitVulkanLayer/DispatchTableTest.cpp

```diff
@@ -0,0 +1,803 @@
+}
+
+TEST(DispatchTable, CanSupportDebugMarkerExtension) {
+  // We can not create an actual VkDevice, but the first bytes of an any dispatchable type in
+  // Vulkan will be a pointer to a dispatch table. This characteristic will be used by our dispatch
+  // table wrapper, so we need to mimic it.
+  VkLayerDispatchTable some_dispatch_table = {};
+  auto device = absl::bit_cast<VkDevice>(&some_dispatch_table);
+  PFN_vkGetDeviceProcAddr next_get_device_proc_addr_function =
+      +[](VkDevice /*instance*/, const char* name) -> PFN_vkVoidFunction {
+    if (strcmp(name, "vkCmdDebugMarkerBeginEXT") == 0) {
+      PFN_vkCmdDebugMarkerBeginEXT begin_function =
+          +[](VkCommandBuffer /*command_buffer*/,
+              const VkDebugMarkerMarkerInfoEXT* /*label_info*/) {};
+      return absl::bit_cast<PFN_vkVoidFunction>(begin_function);
+    }
+    if (strcmp(name, "vkCmdDebugMarkerEndEXT") == 0) {
+      PFN_vkCmdDebugMarkerEndEXT end_function = +[](VkCommandBuffer /*command_buffer*/) {};
+      return absl::bit_cast<PFN_vkVoidFunction>(end_function);
+    }
+    return nullptr;
+  };
+
+  DispatchTable dispatch_table = {};
+  dispatch_table.CreateDeviceDispatchTable(device, next_get_device_proc_addr_function);
+  EXPECT_TRUE(dispatch_table.IsDebugMarkerExtensionSupported(device));
+}
+
+TEST(DispatchTable, CanCallEnumerateDeviceExtensionProperties) {
+  // We can not create an actual VkInstance, but the first bytes of an any dispatchable type in
+  // Vulkan will be a pointer to a dispatch table. This characteristic will be used by our dispatch
+  // table wrapper, so we need to mimic it.
+  VkLayerInstanceDispatchTable some_dispatch_table = {};
+  auto instance = absl::bit_cast<VkInstance>(&some_dispatch_table);
+  PFN_vkGetInstanceProcAddr next_get_instance_proc_addr_function =
+      +[](VkInstance /*instance*/, const char* name) -> PFN_vkVoidFunction {
+    if (strcmp(name, "vkEnumerateDeviceExtensionProperties") == 0) {
+      PFN_vkEnumerateDeviceExtensionProperties function =
+          +[](VkPhysicalDevice /*physical_device*/, const char* /*layer_name*/,
+              uint32_t* property_count, VkExtensionProperties *
+              /*properties*/) -> VkResult {
+        *property_count = 42;
+        return VK_SUCCESS;
+      };
+      return absl::bit_cast<PFN_vkVoidFunction>(function);
+    }
+    return nullptr;
+  };
+
+  DispatchTable dispatch_table = {};
+  dispatch_table.CreateInstanceDispatchTable(instance, next_get_instance_proc_addr_function);
+
+  VkPhysicalDevice device = {};
+  uint32_t property_count;
+
+  VkResult result = dispatch_table.EnumerateDeviceExtensionProperties(instance)(
+      device, nullptr, &property_count, nullptr);
+  EXPECT_EQ(result, VK_SUCCESS);
+  EXPECT_EQ(property_count, 42);
+}
+
+TEST(DispatchTable, CanCallGetPhysicalDeviceProperties) {
+  // We can not create an actual VkInstance, but the first bytes of an any dispatchable type in
```

Comment:
`the first bytes of an any dispatchable 
`->
`the first bytes of any dispatchable `

Reply:
Also: I'd probably put this comment somewhere at the top once instead of repeating it so many times.

Reply:
I added a comment at the very top and removed them here.